tools/rtpsrelay/RtpsRelay.cpp

```diff
@@ -86,16 +137,64 @@ int ACE_TMAIN(int argc, ACE_TCHAR* argv[])
     nic_vertical = ACE_INET_Addr(4444);
   }
 
-  DDS::DomainParticipantFactory_var factory = TheParticipantFactoryWithArgs(argc, argv);
-  DDS::DomainParticipantQos qos;
-  factory->get_default_participant_qos(qos);
+#ifdef OPENDDS_SECURITY
+  if (secure) {
+    if (identity_ca_file.empty()) {
+      ACE_ERROR((LM_ERROR, "ERROR: -IdentityCA is empty\n"));
+      return EXIT_FAILURE;
+    }
+    if (permissions_ca_file.empty()) {
+      ACE_ERROR((LM_ERROR, "ERROR: -PermissionsCA is empty\n"));
+      return EXIT_FAILURE;
+    }
+    if (identity_certificate_file.empty()) {
+      ACE_ERROR((LM_ERROR, "ERROR: -IdentityCertificate is empty\n"));
+      return EXIT_FAILURE;
+    }
+    if (identity_key_file.empty()) {
+      ACE_ERROR((LM_ERROR, "ERROR: -IdentityKey is empty\n"));
+      return EXIT_FAILURE;
+    }
+    if (governance_file.empty()) {
+      ACE_ERROR((LM_ERROR, "ERROR: -Governance is empty\n"));
+      return EXIT_FAILURE;
+    }
+    if (permissions_file.empty()) {
+      ACE_ERROR((LM_ERROR, "ERROR: -Permissions is empty\n"));
+      return EXIT_FAILURE;
+    }
+  }
+#endif
 
+  DDS::DomainParticipantFactory_var factory = TheParticipantFactoryWithArgs(argc, argv);
   if (!factory) {
     ACE_ERROR((LM_ERROR, "(%P|%t) %N:%l ERROR: Failed to initialize participant factory\n"));
     return EXIT_FAILURE;
   }
 
-  DDS::DomainParticipant_var application_participant = factory->create_participant(application_domain, qos, nullptr,
+#if defined(OPENDDS_SECURITY)
```

Comment:
`#ifdef OPENDDS_SECURITY` (multiple places)

Reply:
I don't see a way to consolidate this.